workflow/rules/benchmarking.smk

```diff
@@ -206,6 +206,186 @@ rule plot_strain_call_error:
+    output:
+        contigs=(
+            "results/{date}/assembly/{sample}/megahit-{preset}/{sample}.contigs.fasta"
+        ),
+    wildcard_constraints:
+        preset="std|meta-large|meta-sensitive",
+    log:
+        "logs/{date}/megahit-{preset}/{sample}.log",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+        preset=get_megahit_preset,
+    threads: 8
+    conda:
+        "../envs/megahit.yaml"
+    shell:
+        "(megahit -1 {input.fastq1} -2 {input.fastq2} {params.preset} --out-dir {params.outdir} -f && "
+        "mv {params.outdir}/final.contigs.fa {output.contigs} ) > {log} 2>&1"
+
+
+rule assembly_comparison_trinity:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        "results/{date}/assembly/{sample}/trinity/{sample}.contigs.fasta",
+    log:
+        "logs/{date}/trinity/{sample}.log",
+    params:
+        extra="",
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/trinity.yaml"
+    shell:
+        "(Trinity --left {input.fastq1} --max_memory 16G --right {input.fastq2} --CPU {threads} --seqType fq --output {params.outdir} && "
+        "mv {params.outdir}/Trinity.fasta {output} ) > {log} 2>&1"
+
+
+rule assembly_comparison_velvet:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        "results/{date}/assembly/{sample}/velvet/{sample}.contigs.fasta",
+    log:
+        "logs/{date}/velvet/{sample}.log",
+    params:
+        extra="",
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/velvet.yaml"
+    shell:
+        """
+        velveth {params.outdir} 21 -fastq.gz -shortPaired {input.fastq1} {input.fastq2} > {log} 2>&1
+        velvetg {params.outdir} -ins_length 150 -exp_cov 10 >> {log} 2>&1
+        mv {params.outdir}/contigs.fa {output} >> {log} 2>&1
+        """
+
+
+rule assembly_comparison_spades:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs=(
+            "results/{date}/assembly/{sample}/{spadesflavor}/{sample}.contigs.fasta"
+        ),
+    wildcard_constraints:
+        spadesflavor="spades|rnaviralspades|metaspades|coronaspades",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    log:
+        "logs/{date}/{spadesflavor}/{sample}.log",
+    threads: 8
+    conda:
+        "../envs/spades.yaml"
+    shell:
+        "({wildcards.spadesflavor}.py -1 {input.fastq1} -2 {input.fastq2} -o {params.outdir} -t {threads} && "
+        "mv {params.outdir}/contigs.fasta {output.contigs}) > {log} 2>&1"
+
+
+rule order_contigs_assembly_comparison:
```

Comment:
[Rule Inheritance](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#rule-inheritance) could be a better choice here than duplicate the rules. Even better would be the usage of the original rules.